account_cutoff_prepaid/models/account_cutoff.py

```diff
@@ -0,0 +1,192 @@
+        }
+        return res
+
+    @api.multi
+    def get_prepaid_lines(self):
+        self.ensure_one()
+        aml_obj = self.env['account.move.line']
+        line_obj = self.env['account.cutoff.line']
+        mapping_obj = self.env['account.cutoff.mapping']
+        if not self.source_journal_ids:
+            raise UserError(
+                _("You should set at least one Source Journal."))
+        cutoff_date_str = self.cutoff_date
+        # Delete existing lines
+        self.line_ids.unlink()
+
+        if self.forecast:
+            domain = [
+                ('start_date', '<=', self.end_date),
+                ('end_date', '>=', self.start_date),
+                ('journal_id', 'in', self.source_journal_ids.ids)
+                ]
+        else:
+            domain = [
+                ('start_date', '!=', False),
+                ('journal_id', 'in', self.source_journal_ids.ids),
+                ('end_date', '>', cutoff_date_str),
+                ('date', '<=', cutoff_date_str)
+                ]
+
+        # Search for account move lines in the source journals
+        amls = aml_obj.search(domain)
+        # Create mapping dict
+        mapping = mapping_obj._get_mapping_dict(
+            self.company_id.id,
+            self.cutoff_type
+        )
+
+        # Loop on selected account move lines to create the cutoff lines
+        for aml in amls:
+            line_obj.create(self._prepare_prepaid_lines(aml, mapping))
+        return True
+
+    @api.model
+    def _default_cutoff_account_id(self):
+        account_id = super()._default_cutoff_account_id()
+        cutoff_type = self.env.context.get('cutoff_type')
+        company = self.env.user.company_id
+        if cutoff_type == 'prepaid_revenue':
+            account_id = company.default_prepaid_revenue_account_id.id or False
+        elif cutoff_type == 'prepaid_expense':
+            account_id = company.default_prepaid_expense_account_id.id or False
+        return account_id
+
+    @api.model
+    def create(self, vals):
+        res = super(AccountCutoff, self).create(vals)
+        if not res.source_journal_ids:
+            journals = self.env["account.journal"].search([])
+            res.source_journal_ids = journals
+        return res
+
+
+class AccountCutoffLine(models.Model):
+    _inherit = 'account.cutoff.line'
+
+    move_line_id = fields.Many2one(
+        'account.move.line', string='Account Move Line', readonly=True)
+    move_date = fields.Date(
+        related='move_line_id.date', string='Account Move Date', readonly=True)
+    invoice_id = fields.Many2one(
+        related='move_line_id.invoice_id', string='Invoice', readonly=True)
+    start_date = fields.Date(string='Start Date', readonly=True)
+    end_date = fields.Date(string='End Date', readonly=True)
+    total_days = fields.Integer('Total Number of Days', readonly=True)
+    prepaid_days = fields.Integer(
+        string='Prepaid Days', readonly=True,
```

Comment:
The attribute string is redundant. String parameter equal to name of variable (attribute-string-redundant)
```suggestion
        readonly=True,
```